src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,45 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<Machine> machines = new ArrayList<>();
+
+        if (type == Bulldozer.class) {
+            List<Bulldozer> bulldozers = new BulldozerProducer().get();
+            machines.addAll(bulldozers);
+        } else if (type == Excavator.class) {
+            List<Excavator> excavators = new ExcavatorProducer().get();
+            machines.addAll(excavators);
+        } else if (type == Truck.class) {
+            List<Truck> trucks = new TruckProducer().get();
```

Comment:
The `TruckProducer` class is used here but it seems to be missing from the provided files. Ensure that `TruckProducer` is implemented similarly to `BulldozerProducer` and `ExcavatorProducer` to avoid compilation errors.